Add api for crud operation for practitioners

bibekmaharjan opened the pull request:
## Description

* Created GET /practitioner endpoint to fetch all practitioners.
* Created POST /practitioner endpoint to add a new practitioner.
* Created PUT /practitioner/{practitioner_id} to update a practitioner.
* Created DELETE /practitioner/{practitioner_id} to delete a practitioner.
* Authorized all the APIs for authenticated users.

### Screenshots:
![image](https://user-images.githubusercontent.com/30655695/236374555-6c0ade72-d161-46c6-b7d6-e60feed6cca4.png)

![image](https://user-images.githubusercontent.com/30655695/236374605-0d418c79-3dcc-4800-8712-ee121983fa2a.png)

![image](https://user-images.githubusercontent.com/30655695/236374639-50ddadc1-a5bd-46a3-a4cc-51ace8debf2f.png)

![image](https://user-images.githubusercontent.com/30655695/236374511-ccb9185f-b643-4270-8a77-addcd768cc24.png)

bibekmaharjan committed:
Added authentication for sign in and sign up

bibekmaharjan committed:
Added api for crud operations for practitioners and authenticated them

sbimochan commented on dist/app.js:
```diff
@@ -0,0 +1,15 @@
+"use strict";
```
you can ignore the dist folder itself if you're not serving the page directly from github.

sbimochan commented on src/controllers/auth.ts:
```diff
@@ -0,0 +1,67 @@
+ */
+export const signup = async (req, res) => {
+  // Save User to Database
+  await User.create({
+    email: req.body.email,
+    password: bcrypt.hashSync(req.body.password, 8),
+  })
+    .then(() => {
+      res.send({ message: 'User was registered successfully!' });
+    })
+    .catch((err) => {
+      res.status(500).send({ message: err.message });
+    });
```
Don't mix await and then, use try...catch block and await

sbimochan commented on src/controllers/auth.ts:
```diff
@@ -0,0 +1,67 @@
+      email: req.body.email,
+    },
+  })
+    .then((user) => {
```
same here and other places

sbimochan commented on src/controllers/auth.ts:
```diff
@@ -0,0 +1,67 @@
+      }
+
+      let token = jwt.sign({ id: user.id }, config.secret, {
+        expiresIn: 86400, // 24 hours
```
store it as 24hrInSecond or something in constant file

sbimochan commented on src/models/practitioner.ts:
```diff
@@ -0,0 +1,47 @@
+export default (sequelize, Sequelize) => {
+  const practitioner = sequelize.define(
```
aren't there any optional fields?

sbimochan commented on src/models/practitioner.ts:
```diff
@@ -0,0 +1,47 @@
+      },
+    },
+    {
+      timestamps: false,
```
why false? why not create columns like createdAt and updatedAt?

sbimochan commented on tsconfig.json:
```diff
@@ -0,0 +1,10 @@
+      "outDir": "dist",
+    },
+    "include": ["src/**/*"],
+  }
```
add new EOF line

sbimochan commented:
@bibekmaharjan why were these comments resolved without new commits? And why #3 was pushed with same and fixed codes there? Should it be pushed here instead? It would be easy to review only the new changes.

bibekmaharjan commented:
> @bibekmaharjan why were these comments resolved without new commits? And why #3 was pushed with same and fixed codes there? Should it be pushed here instead? It would be easy to review only the new changes.

Yes dai. Sorry for the inconvinience. I have changed base branch for the pr #3 to this branch for now. Will be creating PR and commits accordingly.
